### PR TITLE
ENG-1134 Components for Nat, Nat8,16,32,64 types

### PR DESCRIPTION
**Pull request by aletex1994**
> Added components in formTypes > NatForms for the creation of these CandyType: 
- CandyNat
- CandyNat8
- CandyNat16
- CandyNat32
- CandyNat64

> Created the file converter.ts in NatForms folder , it contains all the functions to convert the typedValue to the specific CandyType

> Added the array SELECT_OPTIONS to the file constants.ts

> Added the object  VALIDATION_ERRORS to the file constants.ts, it stores all the errors to display when the user enter an invalid CandyType

### CODE REVIEW

File: packages/candy_editor/src/assets/formTypes/NatForms/converters.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { CandyNat, CandyNat8, CandyNat16, CandyNat32, CandyNat64 } from '../../../types';
+
+export function isInRange(num: number, min: number, max: number | bigint): boolean {
+  return num >= min && num <= max;
+}
+
+export function convertToNat(typedValue: string): CandyNat | undefined {
```
Comment (Motokoder) [inline packages/candy_editor/src/assets/formTypes/NatForms/converters.ts:7]: Can you add unit tests for each of these functions?

File: packages/candy_editor/src/constants.ts [hunk 2]
```diff
@@ -1 +1,19 @@
-export const NOT_SELECTED = 'Not selected';
\ No newline at end of file
+export const NOT_SELECTED = 'Not selected';
+export const SELECT_OPTIONS = [
+  { label: 'Text', value: 'Text' },
+  { label: 'Nat', value: 'Nat' },
+  {  label: 'Nat8', value: 'Nat8'},
```
Comment (Motokoder) [inline packages/candy_editor/src/constants.ts:5]: Not formatted with prettier. Please configure VS Code "formatter" and "On Save" settings to automatically format when saving a file.

File: packages/candy_editor/src/constants.ts [hunk 3]
```diff
@@ -1 +1,19 @@
...
+  {  label: 'Nat64', value: 'Nat64'},
+  {  label: 'Nats', value: 'Nats'},
+];
+
+export const VALIDATION_ERRORS = {
+    NAT: 'The value must be a Nat number.',
+    NAT8: 'The value must be a Nat number between 0 and 255',
+    NAT16: 'The value must be a Nat number between 0 and 65535',
+    NAT32: 'The value must be a Nat number between 0 and 4294967295',
+    NAT64: 'The value must be a Nat number between 0 and 18446744073709551615',
```
Comment (Motokoder) [inline packages/candy_editor/src/constants.ts:17]: If the IU does not already cleanly wrap the validation messages, please make sure it does. Thanks.

File: packages/candy_editor/src/constants.ts [hunk 4]
```diff
@@ -1 +1,19 @@
...
+    NATS: 'The value must be a Nat number.',
```
Comment (Motokoder) [inline packages/candy_editor/src/constants.ts:18]: If this is for arrays, please remove for now as those requirements haven't been written yet.

File: packages/candy_editor/src/assets/formTypes/NatForms/natForm.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+export const NatForm = (methods: CandyClassEditor) => {
+  const [name, setName] = useState<string>('');
+  const [value, setValue] = useState<CandyNat>();
+  const [immutable, setImmutable] = useState<boolean>(false);
+  const [isInvalid, setIsInvalid] = useState<boolean>(false);
+
+  const onNameChanged = (typedName: React.ChangeEvent<HTMLInputElement>) => {
+    setName(typedName.target.value);
+  };
+  const onImmutableChanged = () => {
```
Comment (Motokoder) [inline packages/candy_editor/src/assets/formTypes/NatForms/natForm.tsx:16]: Please use newlines between functions for easier reading.

**Review by Motokoder — CHANGES_REQUESTED** [review]

**Comment by aletex1994** [comment]
Added Unit Tests + Requested Changes

**Review by Motokoder — APPROVED** [review]
Nice work!!

**Pull request merged**